# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/vm/RegExpObject.cpp
# Commit: 054a08d13999
# Full Hash: 054a08d139993a42ff2de423e1dab2633baadba6
# Author: Jon Coppeard <jcoppeard@mozilla.com>
# Date: 2022-03-04 04:18:28
# Regressor Bug: 1643586
# File Overlap Count: 1
# Description:
#   Bug 1643586 - Part 5: Replace heap state checks that rely on there being a JSContext for the thread r=jandem
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D140192
# ==============================================================================

diff -r b29a261a06ec -r 054a08d13999 js/src/vm/RegExpObject.cpp
--- a/js/src/vm/RegExpObject.cpp	Thu Mar 03 17:49:17 2022 +0000
+++ b/js/src/vm/RegExpObject.cpp	Thu Mar 03 17:49:18 2022 +0000
@@ -153,18 +153,6 @@
   return false;
 }
 
-static inline bool IsMarkingTrace(JSTracer* trc) {
-  // Determine whether tracing is happening during normal marking.  We need to
-  // test all the following conditions, since:
-  //
-  //   1. During TraceRuntime, RuntimeHeapIsBusy() is true, but the
-  //      tracer might not be a marking tracer.
-  //   2. When a write barrier executes, isMarkingTracer is true, but
-  //      RuntimeHeapIsBusy() will be false.
-
-  return JS::RuntimeHeapIsCollecting() && trc->isMarkingTracer();
-}
-
 static const ClassSpec RegExpObjectClassSpec = {
     GenericCreateConstructor<js::regexp_construct, 2, gc::AllocKind::FUNCTION>,
     GenericCreatePrototype<RegExpObject>,
@@ -556,7 +544,7 @@
 
 void RegExpShared::traceChildren(JSTracer* trc) {
   // Discard code to avoid holding onto ExecutablePools.
-  if (IsMarkingTrace(trc) && trc->runtime()->gc.isShrinkingGC()) {
+  if (trc->isMarkingTracer() && trc->runtime()->gc.isShrinkingGC()) {
     discardJitCode();
   }
 
